src/js/components/Form/Form.js

```diff
@@ -153,19 +153,18 @@ const Form = forwardRef(
         nextComponentValue => {
           if (name) {
             // we have somewhere to put this
+            const nextTouched = { ...touched };
+            nextTouched[name] = true;
+
             if (!touched[name]) {
               // don't update if not needed
-              setTouched(prevTouched => {
-                const nextTouched = { ...prevTouched };
-                nextTouched[name] = true;
-                return nextTouched;
-              });
+              setTouched(nextTouched);
```

Comment:
I seem to recall that we had to use the longer form of `setTouched()` to deal with concurrent updates to values, which I think could come programmatically? I don't remember the details.

But, based on how we are changing `valueState`, I can't see how we couldn't do what you have. Just thought I'd add this comment for reference.

Reply:
I had to make this change because `touch` was not being set in the state on the first input change only on the second.